v3/src/components/graph/hooks/use-plot.ts

```diff
@@ -92,14 +92,12 @@ export const usePlotResponders = (props: IPlotResponderProps) => {
 
   useEffect(function respondToCategorySetChanges() {
     return reaction(() => {
-      return layout.categorySetArrays
-    }, (categorySetsArrays) => {
-      if (categorySetsArrays.length) {
-        startAnimation()
-        callRefreshPointPositions(false)
-      }
+      return dataConfiguration.casesChangeCount
+    }, () => {
+      startAnimation()
+      callRefreshPointPositions(false)
     }, {name: "usePlot.respondToCategorySetChanges"})
-  }, [callRefreshPointPositions, layout.categorySetArrays, startAnimation])
+  }, [callRefreshPointPositions, dataConfiguration.casesChangeCount, startAnimation])
```

Comment:
This is going to dispose/reinstall the reaction when `casesChangeCount` changes, which could prevent the reaction from firing at all (since `fireImmediately` isn't specified). I think what you may want is:
```suggestion
  }, [callRefreshPointPositions, dataConfiguration, startAnimation])
```
which says that the reaction will fire when `casesChangeCount` changes, but won't be disposed/reinstalled.

That said, I'm also curious about the change from `categorySetArrays` to `casesChangeCount`. I take it this is what you meant by:

>Another bug fixed here is that a categorical legend was not responding to changes in the order of its categories brought about by dragging in a different graph. This was fixed by reacting to `dataConfiguration?.casesChangeCount`.

Isn't there a concern that this will fire too often? One could listen to the category set itself if only category changes are of interest.

Finally, if we are sticking with `casesChangeCount` then the name of the function (line 93) and the label (line 99) should be changed from `respondToCategorySetChanges` to something more appropriate.

Reply:
Yes, good call. I find I can use `dataConfiguration.allCategoriesForRoles`. Not sure why I didn't see this when I was struggling with it before. `layout.categorySetArrays` was definitely not working, probably because of lack of observability?